src/flask_pyoidc/pyoidc_facade.py

```diff
@@ -238,18 +241,18 @@ def _token_introspection_request(self, access_token: str):
         oic.extension.message.TokenIntrospectionResponse
             Response object contains result of the token introspection.
         """
-        args = {
+        request = {
             'token': access_token,
-            'client_id': self._client.client_id,
-            'client_secret': self._client.client_secret
+            'token_type_hint': 'access_token'
         }
-        token_introspection_request = self._client_extension.construct_TokenIntrospectionRequest(
-            request_args=args
-        )
+        auth_header = _ClientAuthentication(self._client.client_id, self._client.client_secret)('client_secret_basic',
+                                                                                                request)
         logger.info('making token introspection request')
-        return self._client_extension.do_token_introspection(
-            request_args=token_introspection_request,
-            endpoint=self._client.introspection_endpoint)
+        response = self._provider_configuration.requests_session \
+            .post(self._client.introspection_endpoint, data=request, headers=auth_header) \
```

Comment:
Same as above reason. If we don't let pyoidc to interface with IdP, it becomes our responsibility to comply with openid specs. In future PRs, I would have rather replace existing raw requests to the IdP with the functions implemented in pyoidc. I prefer reusing and wrapping the base code over rewriting the functionality.